### PR TITLE
bpo-37122: Make co->co_argcount represent the total number of positonal arguments in the code object

### PR DESCRIPTION
**Pull request by pablogsal**
Originally proposed by @serhiy-storchaka.

https://bugs.python.org/issue37122

### CODE REVIEW

File: Lib/inspect.py [hunk 1]
```diff
@@ -2136,11 +2132,9 @@ def _signature_from_function(cls, func, skip_bound_arg=True):
     pos_count = func_code.co_argcount
     arg_names = func_code.co_varnames
     posonly_count = func_code.co_posonlyargcount
-    positional_count = posonly_count + pos_count
-    positional_only = tuple(arg_names[:posonly_count])
-    positional = tuple(arg_names[posonly_count:positional_count])
+    positional = tuple(arg_names[:pos_count])
```
Comment (serhiy-storchaka) [inline Lib/inspect.py]: `tuple()` is not needed. `arg_names` is already a tuple.

File: Lib/inspect.py [hunk 2]
```diff
@@ -2136,11 +2132,9 @@ def _signature_from_function(cls, func, skip_bound_arg=True):
...
     keyword_only_count = func_code.co_kwonlyargcount
-    keyword_only = arg_names[positional_count:(positional_count + keyword_only_count)]
+    keyword_only = arg_names[pos_count:(pos_count+ keyword_only_count)]
```
Comment (serhiy-storchaka) [inline Lib/inspect.py]: Parentheses are not needed.

File: Objects/codeobject.c [hunk 3]
```diff
@@ -152,11 +152,10 @@ PyCode_New(int argcount, int posonlyargcount, int kwonlyargcount,
     }
 
     n_varnames = PyTuple_GET_SIZE(varnames);
-    if (posonlyargcount + argcount <= n_varnames
+    if (argcount <= n_varnames
```
Comment (serhiy-storchaka) [inline Objects/codeobject.c]: Can it be merged into a single line?

File: Python/ceval.c [hunk 4]
```diff
@@ -3954,7 +3952,7 @@ _PyEval_EvalCodeWithName(PyObject *_co, PyObject *globals, PyObject *locals,
     }
 
     /* Copy positional only arguments into local variables */
-    if (argcount > co->co_argcount + co->co_posonlyargcount) {
+    if (argcount > co->co_argcount) {
```
Comment (serhiy-storchaka) [inline Python/ceval.c:3955]: It looks to me that these two loops can be merged.

File: Objects/codeobject.c (hunk 3 above)
Comment (serhiy-storchaka) [reply Objects/codeobject.c]: At the beginning of this function add the condition `argcount < posonlyargcount` instead of `argcount < 0`.

**Review by serhiy-storchaka — COMMENTED** [review]
The code looks much clearer now!

**Comment by pablogsal** [comment]
@serhiy-storchaka Done in 09f612c!

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**